src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/appointments/AppointmentsService.kt

```diff
@@ -111,15 +112,21 @@ class AppointmentsService(
   }
 
   suspend fun deleteAppointment(event: AppointmentDomainEvent) {
+    val (appointmentInstanceId) = event.additionalInformation
     val telemetryMap =
-      mutableMapOf("appointmentInstanceId" to event.additionalInformation.appointmentInstanceId.toString())
+      mutableMapOf("appointmentInstanceId" to appointmentInstanceId.toString())
 
     runCatching {
       val nomisEventId =
-        mappingService.getMappingGivenAppointmentInstanceId(event.additionalInformation.appointmentInstanceId).nomisEventId
+        mappingService.getMappingGivenAppointmentInstanceId(appointmentInstanceId).nomisEventId
           .also { telemetryMap["nomisEventId"] = it.toString() }
 
-      nomisApiService.deleteAppointment(nomisEventId)
+      try {
+        nomisApiService.deleteAppointment(nomisEventId)
+      } catch (e: WebClientResponseException.NotFound) {
+        telemetryClient.trackEvent("appointment-delete-missing-ignored", telemetryMap, null)
+      }
+      mappingService.deleteMapping(appointmentInstanceId)
```

Comment:
Also now deleting the mapping. I think I omitted this originally thinking deletions were rare and it might aid troubleshooting to keep the old mapping.

Reply:
I think we would expect nomisApiService.deleteAppointment to be idempotent and not a throw a 404  exception if not found and you are going to ignore.

I think a better approach might have been

1) Check if we have a mapping
2) If not appisnights log the ignore
3) If does delete

Reply:
Confirmed that things are ok as they are for this scenario (as it is handling a problem caused by a missing 2 way sync similar to visit cancellations)